packages/cli/src/eventbus/MessageEventBusDestination/MessageEventBusDestinationSyslog.ee.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import syslog from 'syslog-client';
+import { eventBus } from '../MessageEventBus/MessageEventBus';
+import {
+	MessageEventBusDestinationOptions,
+	MessageEventBusDestinationSyslogOptions,
+	MessageEventBusDestinationTypeNames,
+} from 'n8n-workflow';
+import { MessageEventBusDestination } from './MessageEventBusDestination.ee';
+import { isLogStreamingEnabled } from '../MessageEventBus/MessageEventBusHelper';
+import { EventMessageTypes } from '../EventMessageClasses';
+import { eventMessageGenericDestinationTestEvent } from '../EventMessageClasses/EventMessageGeneric';
+
+export const isMessageEventBusDestinationSyslogOptions = (
+	candidate: unknown,
+): candidate is MessageEventBusDestinationSyslogOptions => {
+	const o = candidate as MessageEventBusDestinationSyslogOptions;
+	if (!o) return false;
+	return o.host !== undefined;
+};
+
+export class MessageEventBusDestinationSyslog
+	extends MessageEventBusDestination
+	implements MessageEventBusDestinationSyslogOptions
+{
+	client: syslog.Client;
+
+	expectedStatusCode?: number;
+
+	host: string;
+
+	port: number;
+
+	protocol: 'udp' | 'tcp';
+
+	facility: syslog.Facility;
+
+	app_name: string;
+
+	eol: string;
+
+	constructor(options: MessageEventBusDestinationSyslogOptions) {
+		super(options);
+		this.__type = options.__type ?? MessageEventBusDestinationTypeNames.syslog;
+		this.label = options.label ?? 'Syslog Server';
+
+		this.host = options.host ?? 'localhost';
+		this.port = options.port ?? 514;
+		this.protocol = options.protocol ?? 'udp';
+		this.facility = options.facility ?? syslog.Facility.Local0;
+		this.app_name = options.app_name ?? 'n8n';
+		this.eol = options.eol ?? '\n';
+		this.expectedStatusCode = options.expectedStatusCode ?? 200;
+
+		this.client = syslog.createClient(this.host, {
```

Comment:
Have we considered using the already installed `winston` library?

Reply:
there's a winston syslog extension but it did not work reliably for some reason (it was rather old as well...)